source/opt/propagator.cpp

```diff
@@ -37,22 +37,28 @@ void SSAPropagator::AddControlEdge(const Edge& edge) {
 }
 
 void SSAPropagator::AddSSAEdges(ir::Instruction* instr) {
-  if (instr->result_id() == 0 || instr->opcode() == SpvOpPhi) {
-    // Ignore instructions that produce no result and Phi instructions. The SSA
-    // edges out of Phi instructions are never added.  Phi instructions can
-    // produce cycles in the def-use web and they are always simulated when a
-    // block is visited.
+  // Ignore instructions that produce no result.
+  if (instr->result_id() == 0) {
     return;
   }
 
   get_def_use_mgr()->ForEachUser(
       instr->result_id(), [this](ir::Instruction* use_instr) {
         // If the basic block for |use_instr| has not been simulated yet, do
-        // nothing.
+        // nothing.  The instruction |use_instr| will be simulated next time the
+        // block is scheduled.
         if (!BlockHasBeenSimulated(ctx_->get_instr_block(use_instr))) {
           return;
         }
 
+        // If |use_instr| is a Phi, ignore this edge.  Phi instructions can form
```

Comment:
I'd probably put this first in the sequence of tests.  It might save some time.

Reply:
Done.